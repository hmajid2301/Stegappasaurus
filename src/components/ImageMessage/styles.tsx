import { StyleSheet } from "react-native";

import { colors, fonts } from "~/modules";

const styles = StyleSheet.create({
  backgroundImage: {
    backgroundColor: "rgba(0, 0, 0, 0.7)",
    height: "100%",
    width: "100%"
  },

  textInputContainer: {
    alignItems: "center",
    flex: 1,
    justifyContent: "center"
  },

  message: {
    backgroundColor: "rgba(0, 0, 0, 0.3)",
    color: "#ffffff",
    fontFamily: fonts.bodyLight,
    fontSize: 22,
    paddingHorizontal: 20,
    paddingVertical: 20
  }
});

export default styles;
