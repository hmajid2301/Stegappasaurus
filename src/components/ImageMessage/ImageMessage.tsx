import * as React from 'react';
import {
  ImageBackground,
  KeyboardAvoidingView,
  TextInput,
  View,
} from 'react-native';
import {
  NavigationEventSubscription,
  NavigationScreenProp,
} from 'react-navigation';

import DismissKeyboard from '~/components/DismissKeyboard';
import {pureWhite} from '~/constants/colors';
import styles from './styles';

interface IProps {
  navigation: NavigationScreenProp<any, any>;
  action?: (message: string) => void;
  editable: boolean;
  message?: string;
  photo: string;
}

interface IState {
  message: string;
}

export default class ImageMessage extends React.Component<IProps, IState> {
  public static defaultProps = {
    action: () => null,
    message: 'Enter your message here',
  };

  private focusListener: NavigationEventSubscription | null;
  private textInput: TextInput | null;

  constructor(props: IProps) {
    super(props);
    this.state = {
      message: '',
    };
    this.textInput = null;
    this.focusListener = null;
  }

  public render() {
    return (
      <KeyboardAvoidingView behavior="height">
        <DismissKeyboard>
          <ImageBackground
            source={{uri: this.props.photo}}
            style={styles.backgroundImage}>
            <View style={styles.textInputContainer}>
              <TextInput
                autoFocus={true}
                blurOnSubmit={true}
                editable={this.props.editable}
                enablesReturnKeyAutomatically={true}
                multiline={true}
                onChangeText={this.updateText}
                onSubmitEditing={
                  this.props.action !== undefined
                    ? this.props.action.bind(this, this.state.message)
                    : () => null
                }
                placeholder={this.props.message}
                placeholderTextColor={pureWhite}
                ref={ref => {
                  this.textInput = ref;
                }}
                style={styles.message}
                underlineColorAndroid="transparent"
              />
            </View>
          </ImageBackground>
        </DismissKeyboard>
      </KeyboardAvoidingView>
    );
  }

  public componentDidMount() {
    this.focusListener = this.props.navigation.addListener('didFocus', () => {
      if (this.textInput !== null) {
        this.textInput.focus();
      }
    });
  }

  public componentWillUnmount() {
    if (this.focusListener) {
      this.focusListener.remove();
    }
  }

  private updateText = (message: string) => {
    this.setState({message});
  };
}
