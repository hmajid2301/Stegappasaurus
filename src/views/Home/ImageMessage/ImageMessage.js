import PropTypes from 'prop-types';
import React, { Component } from 'react';
import {
  ImageBackground,
  KeyboardAvoidingView,
  TextInput,
  View,
} from 'react-native';

import styles from './styles';


export default class ImageMessage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      message: '',
    };
  }

  static propTypes = {
    action: PropTypes.func.isRequired,
    photo: PropTypes.string.isRequired,
  }

  render() {
    return (
      <KeyboardAvoidingView behavior='padding' enabled>
        <ImageBackground source={{ uri: this.props.photo }} style={styles.selectedImage}>
          <View style={styles.textContainer}>
            <TextInput
              autoFocus={true}
              blurOnSubmit={true}
              enablesReturnKeyAutomatically={true}
              multiline={true}
              onChangeText={message => this.setState({ message })}
              onSubmitEditing={this.props.action}
              placeholder='Enter your message here....'
              style={styles.userMessage}
              underlineColorAndroid='transparent'
            />
          </View>
        </ImageBackground>
      </KeyboardAvoidingView>
    );
  }
}
