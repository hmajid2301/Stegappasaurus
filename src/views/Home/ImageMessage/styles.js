import { StyleSheet } from 'react-native';

import { colors, fonts } from '~/util/styles';


const styles = StyleSheet.create({
  textContainer: {
    alignItems: 'center',
    flex: 1,
    justifyContent: 'center',
  },

  selectedImage: {
    backgroundColor: 'rgba(0, 0, 0, 0.7)',
    width: '100%',
    height: '100%',
  },


  userMessage: {
    backgroundColor: 'rgba(0, 0, 0, 0.3)',
    color: colors.pureWhite,
    fontFamily: fonts.body,
    fontSize: 22,
    paddingVertical: 20,
    paddingHorizontal: 20,
  },
});

export default styles;
