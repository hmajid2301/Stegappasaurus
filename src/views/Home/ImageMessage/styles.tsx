import { StyleSheet } from "react-native";

import { colors, fonts } from "~/util/styles";

const styles = StyleSheet.create({
  textContainer: {
    alignItems: "center",
    flex: 1,
    justifyContent: "center"
  },

  selectedImage: {
    backgroundColor: "rgba(0, 0, 0, 0.7)",
    height: "100%",
    width: "100%"
  },

  userMessage: {
    backgroundColor: "rgba(0, 0, 0, 0.3)",
    color: colors.pureWhite,
    fontFamily: fonts.body,
    fontSize: 22,
    paddingHorizontal: 20,
    paddingVertical: 20
  }
});

export default styles;
