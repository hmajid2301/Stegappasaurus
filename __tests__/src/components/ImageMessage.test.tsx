import { mount, shallow } from "enzyme";
import React from "react";

import ImageMessage from "~/components/ImageMessage";

const navigation = {
  addListener: jest.fn()
};

describe("ImageMessage: Match snapshots", () => {
  test("1", () => {
    const component = shallow(
      <ImageMessage
        action={() => null}
        navigation={navigation as any}
        editable={true}
        photo={"images.png"}
      />
    );
    expect(component).toMatchSnapshot();
  });

  test("2", () => {
    const component = shallow(
      <ImageMessage
        action={() => null}
        navigation={navigation as any}
        editable={true}
        message="Hello"
        photo={"images.png"}
      />
    );
    expect(component).toMatchSnapshot();
  });
});

describe("ImageMessag Props", () => {
  test("onChangeText", () => {
    const component = mount(
      <ImageMessage
        action={() => null}
        navigation={navigation as any}
        editable={true}
        photo={"images.png"}
      />
    );

    (component
      .find("TextInput")
      .first()
      .props() as any).onChangeText("Hello");

    expect(component.state("message")).toEqual("Hello");
  });

  test("onSubmitEditting", () => {
    const action = jest.fn();
    const component = mount(
      <ImageMessage
        action={action}
        navigation={navigation as any}
        editable={true}
        photo={"images.png"}
      />
    );

    (component
      .find("TextInput")
      .first()
      .props() as any).onSubmitEditing("Hello");

    expect(action).toHaveBeenCalled();
  });
});
