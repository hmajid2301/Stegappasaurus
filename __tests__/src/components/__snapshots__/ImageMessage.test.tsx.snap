// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageMessage: Match snapshots 1 1`] = `
<KeyboardAvoidingView
  behavior="height"
  enabled={true}
  keyboardVerticalOffset={0}
>
  <DismissKeyboard>
    <ImageBackground
      source={
        Object {
          "uri": "images.png",
        }
      }
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          autoFocus={true}
          blurOnSubmit={true}
          editable={true}
          enablesReturnKeyAutomatically={true}
          multiline={true}
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="Enter your message here"
          placeholderTextColor="#FFF"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.5)",
              "color": "#ffffff",
              "fontFamily": "Roboto-Light",
              "fontSize": 22,
              "height": "100%",
              "paddingHorizontal": 20,
              "paddingVertical": 20,
              "width": "100%",
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
    </ImageBackground>
  </DismissKeyboard>
</KeyboardAvoidingView>
`;

exports[`ImageMessage: Match snapshots 2 1`] = `
<KeyboardAvoidingView
  behavior="height"
  enabled={true}
  keyboardVerticalOffset={0}
>
  <DismissKeyboard>
    <ImageBackground
      source={
        Object {
          "uri": "images.png",
        }
      }
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          autoFocus={true}
          blurOnSubmit={true}
          editable={true}
          enablesReturnKeyAutomatically={true}
          multiline={true}
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="Hello"
          placeholderTextColor="#FFF"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.5)",
              "color": "#ffffff",
              "fontFamily": "Roboto-Light",
              "fontSize": 22,
              "height": "100%",
              "paddingHorizontal": 20,
              "paddingVertical": 20,
              "width": "100%",
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
    </ImageBackground>
  </DismissKeyboard>
</KeyboardAvoidingView>
`;
